build.zig

```diff
@@ -0,0 +1,289 @@
+const std = @import("std");
+const Build = std.Build;
+const Compile = Build.Step.Compile;
+const ResolvedTarget = Build.ResolvedTarget;
+const OptimizeMode = std.builtin.OptimizeMode;
+const builtin = @import("builtin");
+
+const min_supported_ver = "0.15.0";
+comptime {
+    const order = std.SemanticVersion.order;
+    const parse = std.SemanticVersion.parse;
+    if (order(builtin.zig_version, parse(min_supported_ver) catch unreachable) == .lt)
+        @compileError("Box2d requires zig version " ++ min_supported_ver);
+}
+
+pub const Options = struct {
+    shared: bool,
+    unit_tests: bool,
+    disable_simd: bool,
+    avx2: bool,
+
+    const defaults = Options{
+        .shared = false,
+        .unit_tests = false,
+        .disable_simd = false,
+        .avx2 = true,
```

Comment:
Not sure this is desired, or avx2 should always be explicitly enabled. 
Current logic will always enabled avx2 if features is supported by compile target, it will _always_ be false if not supported.